draft-ietf-tls-tls13.md

```diff
@@ -2242,12 +2267,15 @@ extension type (Supported Group Extension):
 The client MUST supply a "supported_groups" extension containing at
 least one group for each key exchange algorithm (currently
 DHE and ECDHE) for which it offers a cipher suite.
-If the client does not supply a "supported_groups" extension with a
-compatible group, the server MUST NOT negotiate a cipher suite of the
-relevant type.  For instance, if a client supplies only ECDHE groups,
-the server MUST NOT negotiate finite field Diffie-Hellman.  If no
-acceptable group can be selected across all cipher suites, then the
-server MUST generate a fatal "handshake_failure" alert.
+Servers receiving a TLS 1.3 ClientHello without this extension yet
+offering either DHE or ECDHE cipher suites MUST send a
+"missing_extension" alert message and close the connection.
```

Comment:
On Thu, Aug 27, 2015 at 10:07 AM, Martin Thomson <notifications@github.com>
wrote:

> In draft-ietf-tls-tls13.md
> <https://github.com/tlswg/tls13-spec/pull/201#discussion_r38119931>:
>
> > @@ -2242,12 +2267,15 @@ extension type (Supported Group Extension):
> >  The client MUST supply a "supported_groups" extension containing at
> >  least one group for each key exchange algorithm (currently
> >  DHE and ECDHE) for which it offers a cipher suite.
> > -If the client does not supply a "supported_groups" extension with a
> > -compatible group, the server MUST NOT negotiate a cipher suite of the
> > -relevant type.  For instance, if a client supplies only ECDHE groups,
> > -the server MUST NOT negotiate finite field Diffie-Hellman.  If no
> > -acceptable group can be selected across all cipher suites, then the
> > -server MUST generate a fatal "handshake_failure" alert.
> > +Servers receiving a TLS 1.3 ClientHello without this extension yet
> > +offering either DHE or ECDHE cipher suites MUST send a
> > +"missing_extension" alert message and close the connection.
>
> I think that the jsep analogy here is bad. In that case, you have
> additional compatibility requirements.
>
> Here, I think that the rule you have is perfect. The server should pick a
> cipher suite, then check named_groups. If there isn't a usable group, barf.
>

If you want to  make this change, please take it to the list.